### PR TITLE
Add event diagnostic counters

### PR DESCRIPTION
**Pull request by peterbryntesson**
Adding event diagnostic counters to console output and Prometheus metrics

### CODE REVIEW

**Review by simonjaeger — APPROVED** [review]
LGTM

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs [hunk 1]
```diff
@@ -216,7 +222,7 @@ private sealed class DataSetWriterSubscription : IDisposable {
                     var sequenceNumber = (uint)Interlocked.Increment(ref _currentSequenceNumber);
                     var snapshot = await Subscription.GetSnapshotAsync().ConfigureAwait(false);
                     if (snapshot != null) {
-                        CallMessageReceiverDelegates(this, sequenceNumber, snapshot);
+                        CallMessageReceiverDelegates(this, sequenceNumber, snapshot, false /* ??? */);
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs:225]: >/* ??? */ [](start = 91, length = 9)

you want to keep the comment?

**Review by mregen — APPROVED** [review]
:shipit:

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs (hunk 1 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs:225]: I'm removing this parameter in a subsequent PR, but yes I will remove the comment.

**Pull request merged**